File: euler/12.c
#include <inttypes.h>
#include <stdio.h>
#include <stdint.h>

uint64_t Tn(uint64_t n)
{
    return (n*(n+1))/2;
}

int factors(uint64_t n)
{
    int c = 0;
    for(uint64_t i = 1; i <= n; i++)
    {
        if((n%i) == 0)
            c++;
    }
    return c;
}

int main(void)
{
    for(int n = 0; 1; n++)
    {
        uint64_t x = Tn(n);
        int c = factors(x);

        if(c > 500)
        {
            printf("%d, %" PRIu64 " %d\n", n, x, c);
            break;
        }
    }
}

File: euler/12.py
def Tn(n):
    return (n*(n+1))//2

def factors(n):
    l = []
    for i in range(1,n+1):
        if (n%i) == 0:
            l.append(i)
    return l

if __name__ == "__main__":
    n = 1
    while True:
        x = Tn(n)
        l = factors(x)

        print(x)
        if len(l) > 500:
            print(n, x, len(l))
            break

        n += 1
